File: programmers/lv1/최소직사각형.py
def solution(sizes):
    for i in range(len(sizes)):
        w = sizes[i][0]
        h = sizes[i][1]
        if w >= h:
            sizes[i][0], sizes[i][1] = sizes[i][1], sizes[i][0]
    wMax = 0
    hMax = 0
    for i in range(len(sizes)):
        wMax = max(sizes[i][0], wMax)
        hMax = max(sizes[i][1], hMax)
    return hMax * wMax

File: programmers/lv1/최소직사각형.js
function solution(sizes) {
  let ans1 = -(10 ** 10);
  let ans2 = -(10 ** 10);
  for (let i of sizes) {
    let maxVal = i[0] > i[1] ? i[0] : i[1];
    let minVal = i[0] > i[1] ? i[1] : i[0];
    ans1 = ans1 < maxVal ? maxVal : ans1;
    ans2 = ans2 < minVal ? minVal : ans2;
  }
  return ans1 * ans2;
}
